apps/explorer/lib/explorer/chain/transaction.ex

```diff
@@ -594,6 +594,16 @@ defmodule Explorer.Chain.Transaction do
   end
 
   # Because there is no contract association, we know the contract was not verified
+  @spec decoded_input_data(
+          %{:__struct__ => Ecto.Association.NotLoaded | Explorer.Chain.Transaction, optional(any()) => any()},
+          any(),
+          any(),
+          any(),
+          any()
+        ) ::
+          {{:error | :ok | binary(), any()}
+           | {:error, :contract_not_verified | :contract_verified, list()}
+           | {:ok, binary(), binary(), list()}, any(), any()}
   def decoded_input_data(tx, skip_sig_provider? \\ false, options, full_abi_acc \\ %{}, methods_acc \\ %{})
```

Comment:
I suggest to change spec to something like this:
```suggestion
  @spec decoded_input_data(
          %NotLoaded{} | Transaction.t(),
          boolean(),
          [Chain.api?()],
          full_abi_acc,
          methods_acc
        ) ::
          {error_type | success_type, full_abi_acc, methods_acc}
        when full_abi_acc: map(),
             methods_acc: map(),
             error_type: {:error, any()} | {:error, :contract_not_verified | :contract_verified, list()},
             success_type: {:ok | binary(), any()} | {:ok, binary(), binary(), list()}
  def decoded_input_data(tx, skip_sig_provider? \\ false, options, full_abi_acc \\ %{}, methods_acc \\ %{})
```

It's not only brings more type information that makes this function safer, but it's more readable in my opinion